### ISSUE TITLE
Transfer issue with standalone transducer : WARNING: Filter out scaler from pretrained dict because of name not found in target dict

### ISSUE THREAD

**Issue by joazoa** [espnet/espnet#4633]
When i try to do transfer learning with latest master and asr_transducer, i am getting these warnings. 
I suspect no weights are transferred.

I am doing:
    --pretrained_model forkmodel.pth \
    --ignore_init_mismatch true \

In this test case case, both configurations are almost identical,  with the exception of a couple of parameters that were commented out in the source config:
      
    auxiliary_ctc_weight: 
    auxiliary_ctc_dropout_rate: 
    auxiliary_lm_loss_weight: 
    auxiliary_lm_loss_smoothing: 

(I expected that commenting them out in the configuration file would lead to default settings).

So if i just try to continue i have: Missing key(s) in state_dict: "ctc_lin.weight", "ctc_lin.bias", "lm_lin.weight", "lm_lin.bias".

Linux 5.4.0-113-generic #127~18.04.1-Ubuntu SMP Wed May 18 15:40:23 UTC 2022 x86_64
python 3.8

Collecting environment information...
PyTorch version: 1.11.0+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.10.2
Libc version: glibc-2.27

Python version: 3.8.0 (default, Dec  9 2021, 17:53:27)  [GCC 8.4.0] (64-bit runtime)
Python platform: Linux-5.4.0-37-generic-x86_64-with-glibc2.27
Is CUDA available: True
CUDA runtime version: 11.2.152

Nvidia driver version: 460.91.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] mypy==0.971
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.3
[pip3] pytest-mypy==0.9.1
[pip3] pytorch-lightning==1.7.2
[pip3] pytorch-ranger==0.1.1
[pip3] pytorch-wpe==0.0.1
[pip3] torch==1.11.0+cu113
[pip3] torch-complex==0.4.3
[pip3] torch-optimizer==0.1.0
[pip3] torch-stoi==0.1.2
[pip3] torchaudio==0.11.0+cu113
[pip3] torchmetrics==0.7.3
[pip3] warprnnt-pytorch==0.1

**Comment by b-flo** [comment]
Hi,

> I suspect no weights are transferred.

Why do you suspect that? The warning you highlight only concern Linear weights and bias for auxiliary CTC and LM loss.
I may be missing some info but everything seems fine here.

> In this test case case, both configurations are almost identical, with the exception of a couple of parameters that were commented out in the source config:

If the pre-trained model was trained with auxiliary tasks modifying source config won't change a thing. We load source model dict and filter out modules and parameters according to the dest config/loading.

**Comment by joazoa** [comment]
Looks like I forgot to copy paste the actual warnings, sorry. 

train.4.log:[bb614b38177d:0/8] 2022-09-11 19:37:23,747 (load_pretrained_model:26) WARNING: Filter out model from pretrained dict because of name not found in target dict
train.4.log:[bb614b38177d:0/8] 2022-09-11 19:37:23,747 (load_pretrained_model:26) WARNING: Filter out reporter from pretrained dict because of name not found in target dict
train.4.log:[bb614b38177d:0/8] 2022-09-11 19:37:23,747 (load_pretrained_model:26) WARNING: Filter out optimizers from pretrained dict because of name not found in target dict
train.4.log:[bb614b38177d:0/8] 2022-09-11 19:37:23,747 (load_pretrained_model:26) WARNING: Filter out schedulers from pretrained dict because of name not found in target dict
train.4.log:[bb614b38177d:0/8] 2022-09-11 19:37:23,748 (load_pretrained_model:26) WARNING: Filter out scaler from pretrained dict because of name not found in target dict

**Comment by b-flo** [comment]
Are they both standalone Transducer model or the version is different between source and target models?
I'll take a look.

**Comment by joazoa** [comment]
Both source and target are standalone transducer models, the only difference in configuration is the auxiliary settings (they are only present in the target).

**Comment by b-flo** [comment]
I just tested with the `AN4` dataset and I don't have any issue on my side. Using a pre-trained model with auxiliary tasks and re-training without ones, only the following parameters are filtered:

```
[calculia] 2022-09-12 21:54:02,216 (load_pretrained_model:26) WARNING: Filter out ctc_lin.weight from pretrained dict because of name not found in target dict                                             
[calculia] 2022-09-12 21:54:02,216 (load_pretrained_model:26) WARNING: Filter out ctc_lin.bias from pretrained dict because of name not found in target dict
[calculia] 2022-09-12 21:54:02,216 (load_pretrained_model:26) WARNING: Filter out lm_lin.weight from pretrained dict because of name not found in target dict                                             
[calculia] 2022-09-12 21:54:02,216 (load_pretrained_model:26) WARNING: Filter out lm_lin.bias from pretrained dict because of name not found in target dict
```

For this issue:

> So if i just try to continue i have: Missing key(s) in state_dict: "ctc_lin.weight", "ctc_lin.bias", "lm_lin.weight", "lm_lin.bias".

This is because you modified the config and tried to resume model training. You need to delete the folder for a complete re-run.

For this one:

```
train.4.log:[bb614b38177d:0/8] 2022-09-11 19:37:23,747 (load_pretrained_model:26) WARNING: Filter out model from pretrained dict because of name not found in target dict
...
train.4.log:[bb614b38177d:0/8] 2022-09-11 19:37:23,748 (load_pretrained_model:26) WARNING: Filter out scaler from pretrained dict because of name not found in target dict
```


I'm not really sure what situation led to such mismatch.

**Comment by joazoa** [comment]
I don't know if it matters, but i started with a model without auxiliary tasks (they were commented out) and tried to retrain with auxiliary tasks, not the other way around.  I will try again tomorrow. 

I didn't consider the other thing an issue, just added it to clarify but made it more confusing instead (especially since i forgot to copy paste the actual issue :/).

**Comment by b-flo** [comment]
> I don't know if it matters, but i started with a model without auxiliary tasks (they were commented out) and tried to retrain with auxiliary tasks, not the other way around. I will try again tomorrow.

I tried and no warning message was displayed.

Was the original model trained with an older ESPnet version compared to your new run? Maybe one commit broke backward compatibility.

**Comment by joazoa** [comment]
Hmm now that you mention it, it might be the case, I will try to train from scratch, fork again and let you know. thank you!

**Comment by b-flo** [comment]
Could you share the commit ID for each version please? If I broke backward compatibility along the way, I need to double check some things.

I'm closing the issue, if there are new things related to the initial post, feel free to re-open it!

**Comment by joazoa** [comment]
The warning was not due to the asr_transducer but due to the source model. I was using the checkpoint instead of the normal .pth  file.
